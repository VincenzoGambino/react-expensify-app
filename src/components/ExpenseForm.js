import React from 'react';
import moment from 'moment';
import { SingleDatePicker } from 'react-dates';

export default class ExpenseForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      description: props.expense ? props.expense.description : '',
      note: props.expense ? props.expense.note : '',
      amount: props.expense ? (props.expense.amount / 100).toString() : '',
      createdAt: props.expense ? moment(props.expense.createdAt) : moment(),
      calendarFocused: true,
      error: ''
    };
  }


  handleDescriptionChange = (e) => {
    const description = e.target.value;
    this.setState(() => ({description}));
  };

  handleAmountChange = (e) => {
    const amount = e.target.value;
    if (!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)) {
      this.setState(() => ({ amount }));
    }
  };

  handleNoteChange = (e) => {
    const note = e.target.value;
    this.setState(() => ({ note }));
  };

  handleDateChange = (createdAt) => {
    if (createdAt) {
      this.setState(() => ({createdAt}));
    }
  };

  handleFocusChange = (({focused}) => {
    this.setState(() => ({calendarFocused: focused}));
  });

  onSubmit = (e) => {
    e.preventDefault();
    if (!this.state.description || !this.state.amount) {
      this.setState(() => ({error: 'Please provide description and amount'}));
    } else {
      this.setState(() => ({error: ''}));
      this.props.onSubmit({
        description: this.state.description,
        amount: parseFloat(this.state.amount, 10) * 100,
        createdAt: this.state.createdAt.valueOf(),
        note: this.state.note
      });
    }
  };
  
  render() {
    return (
      <div>
        {this.state.error && <p>{this.state.error}</p>}
        <form onSubmit={this.onSubmit}>
          <input 
            type="text"
            placeholder='Description'
            autoFocus
            value={this.state.description}
            onChange={this.handleDescriptionChange}
          />
          <input
            type="text"
            placeholder='Amount'
            value={this.state.amount}
            onChange={this.handleAmountChange}
          />
          <SingleDatePicker
            date={this.state.createdAt}
            onDateChange={this.handleDateChange}
            focused={this.state.calendarFocused}
            onFocusChange={this.handleFocusChange}
            numberOfMonths={1}
            isOutsideRange={(day) =>  false}

          />
          <textarea
            placehodlder="Please add note (Optional)"
            value={this.state.note}
            onChange={this.handleNoteChange}
          ></textarea>
          <button>Add Expense</button>
        </form>
      </div>
    );
  }
}