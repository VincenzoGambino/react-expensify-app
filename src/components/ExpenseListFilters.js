import React from 'react';
import { connect } from 'react-redux';
import { setTextFilter, sortBy, setStartDate, setEndDate  } from  '../actions/filters';
import { DateRangePicker } from 'react-dates';

export class ExpenseListFilters extends React.Component {
  state = {
    calendarFocused: null,
  };

  onDatesChange = ({startDate, endDate}) => {
    this.props.setStartDate(startDate);
    this.props.setEndDate(endDate);

  };

  onFocusChange = (calendarFocused) => {
    this.setState(() => ({calendarFocused}));
  }

  onTextChange = ((e) => {
    this.props.setTextFilter(e.target.value);
  });

  onSelectChange = ((e) => {
    this.props.sortBy(e.target.value);
  });
   render() {
    return (
      <div>
        <input type="text" value={this.props.filters.text} onChange={ this.onTextChange }/>
        <select value={this.props.filters.sortBy} onChange={this.onSelectChange}>
          <option value="date">Date</option>
          <option value="amount">Amount</option>
        </select>
        <DateRangePicker
          startDate={this.props.filters.startDate}
          endDate={this.props.filters.endDate}
          onDatesChange={this.onDatesChange}
          focusedInput={this.state.calendarFocused}
          onFocusChange={this.onFocusChange}
          numberOfMonths={1}
          isOutsideRange={() => false}
          showClearDates={true}
        />
      </div>
    );
   }
}

const mapStateToProps = (state) => {
  return {
    filters: state.filters
  }
};

const mapDispatchToProps = (dispatch) => {
  return {
    setTextFilter: (text) => {
      dispatch(setTextFilter(text))
    },
    sortBy: (sortBy) => {
      dispatch(sortBy(sortBy))
    },
    setStartDate: (startDate) => {
      dispatch(setStartDate(startDate))
    },
    setEndDate: (endDate) => {
      dispatch(setEndDate(endDate))
    }
  }
}
export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);