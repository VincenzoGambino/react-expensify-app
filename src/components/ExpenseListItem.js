import React from 'react';
import { Link } from 'react-router-dom';

const ExpenseListItem = ({id, description, amount, createdAt}, props) => {
  return (
    <div>
      <h3><Link to={`/edit/${id}`}>{description}</Link></h3>
      <p>Amount: {amount} - Create at: {createdAt}</p>
    </div>
  );
};

export default ExpenseListItem;