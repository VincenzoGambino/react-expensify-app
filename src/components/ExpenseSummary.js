import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import expenseTotal from '../selectors/expense-total';
import numeral from 'numeral';


export const ExpenseSummary = (props) => {
  const wording = props.expenseCount === 1 ? 'expense' : 'expenses';
  return (
    <div>
    <h1>Viewing {props.expenseCount} {wording} totalling {numeral(props.expenseTotal / 100).format('$0,00.00')}</h1>
     </div>
  );
};

const mapStateToProps = (state) => {
  const expenses = selectExpenses(state.expenses, state.filters);
  return {
    expenseTotal: expenseTotal(expenses),
    expenseCount: expenses.length
  }
};

export default connect(mapStateToProps)(ExpenseSummary);
