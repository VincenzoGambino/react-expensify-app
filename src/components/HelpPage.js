import React from 'react';

// const HelpPage = () => (
//     <div>
//         This is from My Help component
//     </div>
// );

const HelpPage = (props) => {
    return ('Test');
};

export default HelpPage;