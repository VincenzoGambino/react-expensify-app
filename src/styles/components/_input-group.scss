.input-group {
  display:flex;
  margin-bottom: $m-size;
  flex-direction: column;
  @media (min-width: $desktop-breakpoint){
    flex-direction: row;
    margin-bottom: $l-size;
  }
}

.input-group__item {
  margin-left: $s-size;
  @media (min-width: $desktop-breakpoint){
    flex-direction: row;
    margin: 0 $l-size 0 0;
  }
}

.input-text {
  border: 1px solid #cacccd;
  height:50px;
  font-size: $font-size-large;
  font-weight: 300;
  padding: $s-size;
}

.select {
  @extend .input-text; 
}

.textarea {
  @extend .input-text;
  height: 10rem;
}