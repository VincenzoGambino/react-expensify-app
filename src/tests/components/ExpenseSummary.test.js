import React from 'react'
import { shallow } from 'enzyme';
import { ExpenseSummary } from '../../components/ExpenseSummary';

import expenses from '../fixtures/expenses';

test('check with 0 expenses', () => {
  const wrapper = shallow(<ExpenseSummary expenseCount={0} expenseTotal={0}/>);
  expect(wrapper).toMatchSnapshot();
});

test('check with 1 expenses', () => {
  const wrapper = shallow(<ExpenseSummary expenseCount={1} expenseTotal={123}/>);
  expect(wrapper).toMatchSnapshot();
});

test('check with 1 expenses', () => {
  const wrapper = shallow(<ExpenseSummary expenseCount={23} expenseTotal={23400}/>);
  expect(wrapper).toMatchSnapshot();
});