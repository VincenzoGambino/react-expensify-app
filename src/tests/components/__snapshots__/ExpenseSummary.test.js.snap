// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check with 0 expenses 1`] = `
<div>
  <h1>
    Viewing 
    0
     
    expenses
     totalling 
    $0.00
  </h1>
</div>
`;

exports[`check with 1 expenses 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $1.23
  </h1>
</div>
`;

exports[`check with 1 expenses 2`] = `
<div>
  <h1>
    Viewing 
    23
     
    expenses
     totalling 
    $234.00
  </h1>
</div>
`;
